polaris/hub/client.py

```diff
@@ -520,6 +520,13 @@ def upload_dataset(
                 See also: https://www.python-httpx.org/advanced/#timeout-configuration
             owner: Which Hub user or organization owns the artifact. Takes precedence over `dataset.owner`.
         """
+
+        # Check if the specified license for the dataset is supported
+        if dataset.license and dataset.license.id not in SupportedLicenseType:
+            raise InvalidDatasetError(
+                f"\nThe license type you have specified for this dataset is not supported by the Polaris Hub.\nOnly some Creative Commons licenses are supported - {[member.value for member in SupportedLicenseType]}. For more information, see https://creativecommons.org/share-your-work/cclicenses/"
+            )
```

Comment:
This is probably out of scope for this PR, but I'm thinking how we should go about doing the checks before uploading to the Hub (we'll need more) more robustly.

Two ideas:
1. We make the data model for the Dataset more stringent, such that it matches what is expected by the Hub.
2. We create a subclass of the Dataset (and Benchmark and Result) that are more stringent and we validate against this Pydantic model on upload.

Thoughts?

Reply:
Orrrr we keep relying on the errors returned by the Hub to prevent code duplication. This is what we currently do, but the errors have been confusing for some users so we would have to better format then

Reply:
Yeah, agreed. I believe the errors we throw in case there is mismatch between expected and provided data structure, have a JSON field (see [here](https://github.com/polaris-hub/polaris-hub/blob/main/src/lib/errors.ts#L77-L81)) so we should be able to parse and format these more nicely.

Reply:
@Andrewq11 Did you create an issue for error formatting already? We can reference this thread in that issue.